src/shell/commands/cold_backup.cpp

```diff
@@ -143,64 +145,51 @@ bool add_backup_policy(command_executor *e, shell_context *sc, arguments args)
 
 bool ls_backup_policy(command_executor *e, shell_context *sc, arguments args)
 {
-    ::dsn::error_code err = sc->ddl_client->ls_backup_policy();
+    argh::parser cmd(args.argc, args.argv);
+    const bool json = cmd[{"-j", "--json"}];
+
+    ::dsn::error_code err = sc->ddl_client->ls_backup_policy(json);
     if (err != ::dsn::ERR_OK) {
         std::cout << "ls backup policy failed" << std::endl;
-    } else {
-        std::cout << std::endl << "ls backup policy succeed" << std::endl;
     }
     return true;
 }
 
 bool query_backup_policy(command_executor *e, shell_context *sc, arguments args)
 {
-    static struct option long_options[] = {{"policy_name", required_argument, 0, 'p'},
-                                           {"backup_info_cnt", required_argument, 0, 'b'},
-                                           {0, 0, 0, 0}};
-    std::vector<std::string> policy_names;
-    int backup_info_cnt = 3;
+    const std::string query_backup_policy_help = "<policy_name> [-b|--backup_info_cnt] [-j|--json]";
+    argh::parser cmd(args.argc, args.argv);
+    RETURN_FALSE_IF_NOT(cmd.pos_args().size() > 1,
+                        "invalid command, should be in the form of '{}'",
+                        query_backup_policy_help);
 
-    optind = 0;
-    while (true) {
-        int option_index = 0;
-        int c;
-        c = getopt_long(args.argc, args.argv, "p:b:", long_options, &option_index);
-        if (c == -1)
-            break;
-        switch (c) {
-        case 'p': {
-            std::vector<std::string> names;
-            ::dsn::utils::split_args(optarg, names, ',');
-            for (const auto &policy_name : names) {
-                if (policy_name.empty()) {
-                    fprintf(stderr, "invalid, empty policy_name, just ignore\n");
-                    continue;
-                } else {
-                    policy_names.emplace_back(policy_name);
-                }
-            }
-        } break;
-        case 'b':
-            backup_info_cnt = atoi(optarg);
-            if (backup_info_cnt <= 0) {
-                fprintf(stderr, "invalid backup_info_cnt %s\n", optarg);
-                return false;
-            }
-            break;
-        default:
-            return false;
+    std::vector<std::string> policy_names;
+    std::vector<std::string> names;
+    ::dsn::utils::split_args(cmd(1).str().c_str(), names, ',');
+    for (const auto &policy_name : names) {
+        if (policy_name.empty()) {
```

Comment:
Empty token will be ignored automatically, see
https://github.com/apache/incubator-pegasus/blob/master/src/utils/strings.h#L74